src/test/ui/proc-macro/invalid-punct-ident-1.rs

```diff
@@ -1,4 +1,5 @@
 // aux-build:invalid-punct-ident.rs
+// needs-unwind if a proc macro panics then the compiler will abort https://github.com/rust-lang/rust/issues/82320
```

Comment:
I find this comment confusing because it doesn't qualify itself (e.g. "if a proc macro panics with panic=abort"). Nit, the wording also insinuates to me that the compiler has a choice in the matter (e.g. "the compiler will abort" instead of "it will abort the compiler") :).

Proc macros can panic to report errors, and because they are run in the compiler process they must be built with `panic=unwind` for this functionality to work. None of this is really a bug that's going to be fixed anytime soon (the "fixes" proposed in that issue are for the compiler to error/warn if you try to build a proc macro with `panic=abort`, for example), so I actually think pasting an issue link is a bit misleading in this case. All of that to say, maybe a terser comment like "uses proc macros" is a better choice.

It would be nice if we could teach compiletest to mark tests that use proc macros like this automatically, but I'm not sure that's so easy.

Reply:
This is very helpful, thank you! To expand a bit in the comment, is `"//needs-unwind uses proc macros, which require unwind to report errors"` a solid way of wording?

Reply:
Actually taking a second look, this only is the set of proc macro tests that actually panic to report errors (there is another API for reporting them that is more common now). So a more accurate wording might be "proc macro panics to report errors" or similar.